apps/vc-api/src/vc-api/credentials/credentials.service.ts

```diff
@@ -177,12 +166,14 @@ export class CredentialsService implements CredentialVerifier {
     vp: VerifiablePresentationDto,
     options: VerifyOptionsDto
   ): Promise<VerificationResultDto> {
-    const verifyOptions: ISpruceVerifyOptions = options;
-
-    return didKitExecutor<VerificationResultDto>(
-      () => verifyPresentation(JSON.stringify(vp), JSON.stringify(verifyOptions)),
-      'verifyPresentation'
+    const w3cVerifyPresentationOptions: W3cVerifyPresentationOptions = {
+      presentation: JsonTransformer.fromJSON(vp, W3cJsonLdVerifiablePresentation),
```

Comment:
A note that as this `vp.verifiableCredential === undefined` exception needs to be handles in several places (3 in total I believe), I think it would probably make sense to extract the `fromJSON` call for VP into a reusable function (perhaps a private method in the service class)